src/rustup.rs

```diff
@@ -104,33 +127,55 @@ pub fn available_targets(toolchain: &str, msg_info: &mut MessageInfo) -> Result<
     })
 }
 
-pub fn install_toolchain(toolchain: &str, msg_info: &mut MessageInfo) -> Result<()> {
+pub fn install_toolchain(toolchain: &QualifiedToolchain, msg_info: &mut MessageInfo) -> Result<()> {
     rustup_command(msg_info, false)
-        .args(&["toolchain", "add", toolchain, "--profile", "minimal"])
+        .args(&[
```

Comment:
Another nit, but if the variable is stored before, this becomes way more legible on a single line:

```rust
pub fn install_toolchain(toolchain: &QualifiedToolchain, msg_info: &mut MessageInfo) -> Result<()> {
    let toolchain = toolchain.to_string();
    rustup_command(msg_info, false)
        .args(&["toolchain", "add", &toolchain, "--profile", "minimal"])
        .run(msg_info, false)
        .wrap_err_with(|| format!("couldn't install toolchain `{toolchain}`"))
}
```